cw-relayer/cmd/price-feeder.go

```diff
@@ -0,0 +1,186 @@
+	"github.com/spf13/cobra"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/ojo-network/cw-relayer/config"
+	"github.com/ojo-network/cw-relayer/oracle"
+	oracleclient "github.com/ojo-network/cw-relayer/oracle/client"
+)
+
+const (
+	logLevelJSON = "json"
+	logLevelText = "text"
+
+	flagLogLevel  = "log-level"
+	flagLogFormat = "log-format"
+
+	envVariablePass = "CW_RELAYER_PASS"
+	accPrefix       = "wasm"
+)
+
+var rootCmd = &cobra.Command{
+	Use:   "cw-relayer [config-file]",
+	Args:  cobra.ExactArgs(1),
+	Short: "cw-relayer is a side-car process for providing Wasm-enabled chains with Ojo's pricing Data",
+	Long:  `cw-relayer is a side-car process for providing Wasm-enabled chains with Ojo's pricing Data.`,
+	RunE:  cwRelayerCmdHandler,
+}
+
+func init() {
+	rootCmd.PersistentFlags().String(flagLogLevel, zerolog.InfoLevel.String(), "logging level")
+	rootCmd.PersistentFlags().String(flagLogFormat, logLevelText, "logging format; must be either json or text")
+
+	rootCmd.AddCommand(getVersionCmd())
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func cwRelayerCmdHandler(cmd *cobra.Command, args []string) error {
+	logLvlStr, err := cmd.Flags().GetString(flagLogLevel)
+	if err != nil {
+		return err
+	}
+
+	logLvl, err := zerolog.ParseLevel(logLvlStr)
+	if err != nil {
+		return err
+	}
+
+	logFormatStr, err := cmd.Flags().GetString(flagLogFormat)
+	if err != nil {
+		return err
+	}
+
+	var logWriter io.Writer
+	switch strings.ToLower(logFormatStr) {
+	case logLevelJSON:
+		logWriter = os.Stderr
+
+	case logLevelText:
+		logWriter = zerolog.ConsoleWriter{Out: os.Stderr}
+
+	default:
+		return fmt.Errorf("invalid logging format: %s", logFormatStr)
+	}
+
+	logger := zerolog.New(logWriter).Level(logLvl).With().Timestamp().Logger()
+
+	cfg, err := config.ParseConfig(args[0])
+	if err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithCancel(cmd.Context())
+	g, ctx := errgroup.WithContext(ctx)
+
+	// listen for and trap any OS signal to gracefully shutdown and exit
+	trapSignal(cancel, logger)
+	//
+	rpcTimeout, err := time.ParseDuration(cfg.RPC.RPCTimeout)
+	if err != nil {
+		return fmt.Errorf("failed to parse RPC timeout: %w", err)
+	}
+
+	// Gather pass via env variable || std input
+	keyringPass, err := getKeyringPassword()
+	if err != nil {
+		return err
+	}
+	//
+	//// client for interacting with the ojo & wasmd chain
+	client, err := oracleclient.NewOracleClient(
+		ctx,
+		logger,
+		cfg.Account.ChainID,
+		cfg.Keyring.Backend,
+		cfg.Keyring.Dir,
+		keyringPass,
+		cfg.RPC.TMRPCEndpoint,
+		rpcTimeout,
+		cfg.Account.Address,
+		cfg.RPC.GRPCEndpoint,
+		cfg.GasAdjustment,
+	)
+	if err != nil {
+		return err
+	}
+
+	newOracle := oracle.New(logger, client, cfg.ContractAddress, cfg.MissedThreshold, cfg.QueryRPC)
```

Comment:
can we call this something instead of "oracle"? Relayer or something?

let's also change the `oracle` folder to `relayer`